src/jobservice/job/impl/systemartifact/cleanup.go

```diff
@@ -0,0 +1,44 @@
+package systemartifact
+
+import (
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/pkg/systemartifact"
+)
+
+type Cleanup struct {
+	sysArtifactManager systemartifact.Manager
+}
+
+func (c *Cleanup) MaxFails() uint {
+	return 1
+}
+
+func (c *Cleanup) MaxCurrency() uint {
+	return 0
+}
+
+func (c *Cleanup) ShouldRetry() bool {
+	return true
```

Comment:
refer to this https://github.com/goharbor/harbor/pull/16833#discussion_r873351759